sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReceiversPumpingScheduler.java

```diff
@@ -0,0 +1,189 @@
+        }
+        ReceiversPumpingScheduler scheduler = INSTANCE.get();
+        if (scheduler != null) {
+            return scheduler;
+        } else {
+            scheduler = new ReceiversPumpingScheduler();
+            if (!INSTANCE.compareAndSet(null, scheduler)) {
+                scheduler.inner.dispose();
+            }
+            return INSTANCE.get();
+        }
+    }
+
+    @Override
+    public Disposable schedule(Runnable task) {
+        return inner.schedule(task);
+    }
+
+    @Override
+    public Disposable schedule(Runnable task, long delay, TimeUnit unit) {
+        return inner.schedule(task, delay, unit);
+    }
+
+    @Override
+    public Disposable schedulePeriodically(Runnable task, long initialDelay, long period, TimeUnit unit) {
+        return inner.schedulePeriodically(task, initialDelay, period, unit);
+    }
+
+    @Override
+    public Worker createWorker() {
+        return inner.createWorker();
+    }
+
+    @Override
+    public long now(TimeUnit unit) {
+        return inner.now(unit);
+    }
+
+    /**
+     *  @deprecated replaced by init.
+     */
+    @Deprecated
+    @Override
+    public void start() {
+        inner.start();
+    }
+
+    @Override
+    public void init() {
+        inner.init();
+    }
+
+    @Override
+    public void dispose() {
+        // NOP as the Scheduler INSTANCE is common.
+    }
+
+    @Override
+    public boolean isDisposed() {
+        return inner.isDisposed();
+    }
+
+    @Override
+    public String toString() {
+        return NAME;
+    }
+
+    private ReceiversPumpingScheduler() {
+        final Supplier<Integer> poolMaxSizeDefault = () -> 20 * Runtime.getRuntime().availableProcessors();
+        // Note: It would be nice to read the custom pool size using com.azure.core.util.Configuration.getGlobalConfiguration.
+        // It requires adding the key to the azure-core known configuration properties, let's evaluate it separately.
+        final Optional<Integer> poolMaxSizeOverridden = Optional.ofNullable(System.getProperty("com.azure.core.amqp.receiversPumpingThreadPoolMaxSize"))
+            .map(m -> {
+                try {
+                    return Integer.parseInt(m);
+                } catch (NumberFormatException e) {
+                    // Use poolMaxSizeDefault (the initialization log below hints the size is chosen).
+                    return null;
+                }
+            });
+        final int poolMaxSize = poolMaxSizeOverridden.orElseGet(poolMaxSizeDefault);
+        this.inner = Schedulers.newBoundedElastic(poolMaxSize, TASK_QUEUE_CAP, NAME, IDLE_TTL_SECONDS, true);
+        (new ClientLogger(ReceiversPumpingScheduler.class)).atVerbose()
+            .log("Initialized common ReceiversPumpingScheduler(maxThreads={})", poolMaxSize);
+    }
+
+    private static boolean isVTSchedulerMode() {
+        // Check any of the Reactor's common (cached) Scheduler to see if "VirtualTimeScheduler" (VTScheduler) is injected;
```

Comment:
I would promote this to the javadoc rather than just a comment. (Mostly because I like that javadoc gets rendered, so it's easier for me to read.)

Reply:
good thought, let's do it.